insights/specs/datasources/system_user_dirs.py

```diff
@@ -0,0 +1,79 @@
+"""
+Custom datasource for checking non-root owned packaged directories
+
+Author: Florian Festi <ffesti@redhat.com>
+"""
+import json
+import stat
+
+from insights import datasource
+from insights.core.spec_factory import DatasourceProvider
+
+
+def get_shells():
+    with open('/etc/shells') as f:
+        return set((line.strip() for line in f))
+
+
+def get_users():
+    shells = get_shells()
+    users = set()
+
+    with open('/etc/passwd') as f:
+        for line in f:
+            entry = line.split(":")
+            if len(entry) < 7:
+                continue
+            shell = entry[6].strip()
+            if shell not in shells or shell in ('/sbin/nologin', '/bin/nologin'):
+                continue
+            users.add(entry[0])
+    users.discard("root")
+    return users
+
+
+def get_groups(users):
+    groups = set()
+
+    with open('/etc/group') as f:
```

Comment:
Would it be possible to use the grp python module to query group info also?

Reply:
I will look at it, thank you.